packages/commonwealth/client/scripts/views/modals/TradeTokenModel/TradeTokenForm/useBuyTrade.ts

```diff
@@ -0,0 +1,193 @@
+  const baseCurrencyBuyAmountDecimals =
+    parseFloat(baseCurrencyBuyAmountString) || 0;
+
+  const { linkSpecificAddressToSpecificCommunity } = useJoinCommunity();
+
+  const { data: ethToCurrencyRateData, isLoading: isLoadingETHToCurrencyRate } =
+    useFetchTokenUsdRateQuery({
+      tokenSymbol: 'ETH',
+      enabled,
+    });
+  const ethToCurrencyRate = parseFloat(
+    ethToCurrencyRateData?.data?.data?.amount || '0.00',
+  );
+  const ethBuyAmount = baseCurrencyBuyAmountDecimals / ethToCurrencyRate;
+  const commonPlatformFeeForBuyTradeInEth =
+    (commonFeePercentage / 100) * ethBuyAmount;
+
+  // imp: this query uses CommunityStakes helper to get eth price, but its
+  // a generic query so no need to initiate a separate Launchpad helper
+  const {
+    data: selectedAddressEthBalance = `0.0`,
+    isLoading: isLoadingUserEthBalance,
+  } = useGetUserEthBalanceQuery({
+    chainRpc: tokenCommunity?.ChainNode?.url || '',
+    ethChainId: tokenCommunity?.ChainNode?.eth_chain_id || 0,
+    walletAddress: selectedAddress || '',
+    apiEnabled: !!(selectedAddress && tokenCommunity && enabled),
+  });
+
+  const {
+    data: unitEthToTokenBuyExchangeRate = 0,
+    isLoading: isLoadingUnitEthToTokenBuyExchangeRate,
+  } = useTokenEthExchangeRateQuery({
+    chainRpc: chainNode.url,
+    ethChainId: chainNode.ethChainId || 0,
+    mode: 'buy',
+    tokenAmount: 1 * 1e18, // convert to wei - get exchange rate of 1 unit token to eth
+    tokenAddress: tradeConfig.token.token_address,
+    enabled: !!(
+      chainNode?.url &&
+      chainNode?.ethChainId &&
+      selectedAddress &&
+      tokenCommunity &&
+      enabled
+    ),
+  });
+
+  const { mutateAsync: buyToken, isLoading: isBuyingToken } =
+    useBuyTokenMutation();
+
+  const { mutateAsync: createTokenTrade, isLoading: isCreatingTokenTrade } =
+    useCreateTokenTradeMutation();
+
+  const onBaseCurrencyBuyAmountChange = (
+    change: React.ChangeEvent<HTMLInputElement> | TokenPresetAmounts,
+  ) => {
+    if (typeof change == 'number') {
+      setBaseCurrencyBuyAmountString(`${change}`);
+    } else if (typeof change == 'string') {
+      // not handling string type preset amounts atm
+    } else {
+      const value = change.target.value;
+
+      if (value === '') setBaseCurrencyBuyAmountString('0');
+      // verify only numbers with decimal (optional) are present
+      else if (/^\d*\.?\d*$/.test(value)) setBaseCurrencyBuyAmountString(value);
+    }
+  };
+
+  const handleTokenBuy = async () => {
+    try {
+      // this condition wouldn't be called, but adding to avoid typescript issues
+      if (
+        !chainNode?.url ||
+        !chainNode?.ethChainId ||
+        !selectedAddress ||
+        !tokenCommunity
+      ) {
+        return;
+      }
+
+      // buy token on chain
+      const payload = {
+        chainRpc: chainNode.url,
+        ethChainId: chainNode.ethChainId,
+        amountEth: ethBuyAmount * 1e18, // amount in wei
```

Comment:
@rbennettcw is this a case for using big int?

Reply:
I discussed with Ian and we need to convert buy amounts to bigint for passing to helper. Resolving this for merge.